### ISSUE TITLE
ValidateCustomerExistsAttribute is not a good pattern to use.

### ISSUE THREAD

**Issue by michael-freidgeim-webjet** [JasonGT/SpaWithAspNetCoreDemo#1]
Hi Jason, 
Thank you for your very useful presentation on Mebourne DDD. 
I've looked at [ValidateCustomerExistsAttribute](https://github.com/JasonGT/SpaWithAspNetCoreDemo/blob/cc9555d4b5f4ecf9923a19e053ebd15b125427e7/Northwind.Presentation/Filters/ValidateCustomerExistsAttribute.cs) and understand, that Asp.Net Core framework injects context object to the attribute. 
However you can't garrantee that in
 ```
  [HttpDelete("{id}")]
        [ValidateCustomerExists]
        public async Task<IActionResult> DeleteCustomer([FromRoute] string id)
        {
            await _deleteCustomerCommand.Execute(id);

            return NoContent();
        }
```
ValidateCustomerExists and _deleteCustomerCommand use the same context and refer to the same database. If your application will support multiple customer databases, it could happen that you check existence in one database, byt try to delete in another.
From different perspective, controller is initiated per request, but attribute is a singleton.
 
Implementing private common method ValidateCustomerExists in CustomersController class and call it from a few methods will be more **simple**, than creating separate attribute. 
In general it's arguable, that common functionality within one class can be considered as "Cross-cutting concerns"
I suggest you to find a better example for cross-cutting concerns.

**Comment by MaximBalaganskiy** [comment]
Correct, the context should be resolved in the execute method. Similar to
middleware

**Comment by MaximBalaganskiy** [comment]
You might also might get concurrency exceptions when several parallel requests trigger the attributes sharing the same constructor injected context
